packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -104,22 +123,103 @@ export abstract class DeployRetrieveExecutor<
   protected abstract postOperation(
     result: DeployRetrieveResult | undefined
   ): Promise<void>;
+  protected abstract handleSourceConflictError(e: any): void;
 }
 
 export abstract class DeployExecutor<T> extends DeployRetrieveExecutor<T> {
   protected async doOperation(
     components: ComponentSet,
     token: vscode.CancellationToken
   ): Promise<DeployResult | undefined> {
+    const projectPath = getRootWorkspacePath();
+    const connection = await WorkspaceContext.getInstance().getConnection();
+
+    const sourceTracking = await SourceTrackingService.createSourceTracking(
+      projectPath,
+      connection
+    );
+    await sourceTracking.ensureLocalTracking();
```

Comment:
In my observances during dev, if this call was made inside utils/SourceTrackingService#createSourceTracking it did not work - meaning newly created files would not be picked up.  When I put the call here in core/baseDeployRetrieve it does work to pick up newly created files.  The "new Apex Class" scenario should be thoroughly tested before merge of this PR.

Reply:
Can you give an example of the test flow you're concerned about for this?

Reply:
Yep, repo steps are included in the PR description up above (buried in my checklist of todos):

 investigate fix potential issue with stl cacheing and possibly not recognizing new files
Repro:
Build this branch, launch Extension Host
Create a New sfdx project
Auth a DevHub
Auth a Scratch Org
Create a New Apex Class
Deploy the class to the scratch org
Modify the class locally
Modify the class on the server
Deploy the class to the scratch org
Expected:
Conflict
Actual:
class is deployed and overwrites the version on the server